src/libstore/binary-cache-store.cc

```diff
@@ -385,7 +385,10 @@ StorePath BinaryCacheStore::addToStore(const string & name, const Path & srcPath
         h = hashString(hashAlgo, s);
     }
 
-    ValidPathInfo info(makeFixedOutputPath(method, *h, name));
+    ValidPathInfo info {
+        makeFixedOutputPath(method, *h, name),
+        Hash::dummy, // Will be fixed in addToStore, which recomputes nar hash
```

Comment:
This is rather tricky. Because the other one calculates the Nar hash, I think we should eventually make an `addToStoreFromDump` out of it, and then this tricky bit can go away.